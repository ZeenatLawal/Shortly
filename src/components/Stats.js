import React from 'react';
import brand from '../images/icon-brand-recognition.svg';
import records from '../images/icon-detailed-records.svg';
import customize from '../images/icon-fully-customizable.svg';

const Stats = () => (
  <>
    <section>
      <h3>Advanced Statistics</h3>
      <p>
        Track how your links are performing across the
        web with our advanced statistics dashboard.
      </p>
      <div>
        <div>
          <img src={brand} alt="Graph icon" />
          <h4>Brand recognition</h4>
          <p>
            Boost your brand recognition with each click. Generic links don&apos;t mean a thing.
            Branded links help instill confidence in your content.
          </p>
        </div>
        <div>
          <img src={records} alt="Records icon" />
          <h4>Detailed Records</h4>
          <p>
            Gain insights into who is clicking your links. Knowing when and where people engage with
            your content helps inform better decisions.
          </p>
        </div>
        <div>
          <img src={customize} alt="Pens icon" />
          <h4>Fully Customizable</h4>
          <p>
            Improve brand awareness and content discoverability through customizable links,
            supercharging audience engagement.
          </p>
        </div>
      </div>
    </section>
    <section>
      <h3>Boost your links today</h3>
      <button type="button">Get Started</button>
    </section>
  </>
);

export default Stats;
