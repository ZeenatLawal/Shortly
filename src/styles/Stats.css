.stats {
  background-color: var(--light-violet);
  padding-top: 4rem;
}

.line {
  height: 0.6rem !important;
  width: 8.5rem;
  background-color: var(--cyan);
  opacity: 1;
  transform: rotate(90deg);
  margin-left: 5.5rem;
}

.icon {
  background-color: var(--dark-violet);
  width: 4.5rem;
  height: 4.5rem;
  padding: 1rem;
  margin: 0 auto;
  position: absolute;
  top: -2rem;
  left: 7.5rem;
}

.stats-card {
  background-color: white;
  padding: 4rem 1rem 1rem 1rem;
  margin-top: 1rem;
}

.stats-card-title {
  color: var(--darker-violet);
  font-weight: 700;
  padding-bottom: 0.5rem;
}

.boost {
  background-color: var(--dark-violet);
  background-image: url('../images/bg-boost-mobile.svg');
  background-repeat: no-repeat;
  background-position: left;
  padding: 5rem 1rem;
}

.boost-title {
  font-weight: 700;
}

@media screen and (min-width: 1024px) {
  .stats-title {
    font-size: 2rem;
  }

  .stats-card {
    width: 19rem;
  }

  .line {
    width: 2rem;
    margin-left: 1rem;
    transform: rotate(180deg);
    margin-top: 10rem;
  }

  .icon {
    top: -1rem;
    left: 2rem;
  }

  .records {
    padding-top: 2.5rem;
  }

  .customize {
    padding-top: 5rem;
  }

  .records .icon {
    top: 1.5rem;
  }

  .customize .icon {
    top: 3.5rem;
  }

  .boost {
    background-image: url('../images/bg-boost-desktop.svg');
    padding: 4rem 1rem;
  }
}
