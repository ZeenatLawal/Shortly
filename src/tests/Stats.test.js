import renderer from 'react-test-renderer';
import Stats from '../components/Stats';

describe('Stats component', () => {
  test('Snapshot test', () => {
    const stats = renderer.create(
      <Stats />,
    )
      .toJSON();
    expect(stats).toMatchSnapshot();
  });
});
