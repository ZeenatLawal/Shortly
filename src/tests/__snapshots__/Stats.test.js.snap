// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stats component Snapshot test 1`] = `
Array [
  <section>
    <h3>
      Advanced Statistics
    </h3>
    <p>
      Track how your links are performing across the web with our advanced statistics dashboard.
    </p>
    <div>
      <div>
        <img
          alt="Graph icon"
          src="icon-brand-recognition.svg"
        />
        <h4>
          Brand recognition
        </h4>
        <p>
          Boost your brand recognition with each click. Generic links don't mean a thing. Branded links help instill confidence in your content.
        </p>
      </div>
      <div>
        <img
          alt="Records icon"
          src="icon-detailed-records.svg"
        />
        <h4>
          Detailed Records
        </h4>
        <p>
          Gain insights into who is clicking your links. Knowing when and where people engage with your content helps inform better decisions.
        </p>
      </div>
      <div>
        <img
          alt="Pens icon"
          src="icon-fully-customizable.svg"
        />
        <h4>
          Fully Customizable
        </h4>
        <p>
          Improve brand awareness and content discoverability through customizable links, supercharging audience engagement.
        </p>
      </div>
    </div>
  </section>,
  <section>
    <h3>
      Boost your links today
    </h3>
    <button
      type="button"
    >
      Get Started
    </button>
  </section>,
]
`;
